src/pint/utils.py

```diff
@@ -1686,6 +1694,103 @@ def remove_dummy_distance(c):
         return c
 
 
+def info_string(prefix_string="# ", comment=None):
+    """Returns an informative string about the current state of PINT.
+
+    Adds:
+    
+    * Creation date
+
+    * PINT version
+
+    * Username
+
+    * Host
+
+    * OS
+
+    * plus a user-supplied comment (if present).
+
+    Username is given by the ``gitpython`` global configuration ``user.name``
+    (if available), otherwise ``getpass.getuser()``.
+
+    Parameters
+    ----------
+    prefix_string: str, default='# '
+        a string to be prefixed to the output (often to designate as a
+        comment or similar)
+    comment: str, optional
+        a free-form comment string to be included if present
+    
+    Returns
+    -------
+    s : str
+        informative string
+
+    Examples
+    --------
+    >>> import pint.utils
+    >>> print(pint.utils.info_string(prefix_string="# ",comment="Example comment"))
+    # Created: 2021-07-21T09:39:45.606894
+    # PINT_version: 0.8.2+311.ge351099d
+    # User: David Kaplan (dlk)
+    # Host: margle-2.local
+    # OS: macOS-10.14.6-x86_64-i386-64bit
+    # Comment: Example comment
+    
+    Multi-line comments are allowed:
+
+    >>> import pint.utils
+    >>> print(pint.utils.info_string(prefix_string="C ",
+    ...                              comment="Example multi-line comment\\nAlso using a different comment character"))
+    C Created: 2021-07-21T09:40:34.172333
+    C PINT_version: 0.8.2+311.ge351099d
+    C User: David Kaplan (dlk)
+    C Host: margle-2.local
+    C OS: macOS-10.14.6-x86_64-i386-64bit
+    C Comment: Example multi-line comment
+    C Comment: Also using a different comment character    
+
+    Notes
+    -----
+    This can be called via ``t.write_TOA_file("combined.tim",comment="trying combination")``
```

Comment:
I like this! Maybe sneak in a parenthesized :func:`~pint.toa.TOAs.write_TOA_file` to get a clickable link?

Reply:
I sort of get what you are looking for, but the syntax is eluding me, and the testing cycle takes a while.  any concrete pointers to a working example?

Reply:
Oh, sorry, that's markdown confusing things:
```
this looks like a live link to the function name: :func:`~pint.toa.TOAs.write_TOA_file`; the tilde shortens the link so it just looks like write_TOA_file() but still links to the docs for that function. 
```
There should be examples various places under docs/, a grep for :func: would pull them up.